src/pages/cart.jsx

```diff
@@ -23,13 +23,97 @@ import {
 import { Seo } from "../components/seo"
 
 export default function CartPage() {
-  const { checkout, loading } = React.useContext(StoreContext)
+  const { checkout, loading, client, setCheckout } = React.useContext(StoreContext)
   const emptyCart = checkout.lineItems.length === 0
 
-  const handleCheckout = () => {
+  const handleCheckout = async () => {
+    if(window.onwardApp) {
+      await window.onwardApp.updateCartForCheckout();
```

Comment:
before going to checkout us the `updateCartForCheckout` function from onward to add insurance. We normally use `fixInsuranceAndProceed`, but since this is a client side app working with a checkout if we come back to the page and it doesn't execute anything the app checkout state might not be updated. `updateCartForCheckout` always clears the cart first so it works a bit bette for this case.